src/runtime/virtcontainers/qemu_test.go

```diff
@@ -76,40 +76,112 @@ func TestQemuKernelParameters(t *testing.T) {
 }
 
 func TestQemuCreateVM(t *testing.T) {
-	qemuConfig := newQemuConfig()
 	assert := assert.New(t)
 
 	store, err := persist.GetDriver()
 	assert.NoError(err)
-	q := &qemu{
-		config: HypervisorConfig{
-			VMStorePath:  store.RunVMStoragePath(),
-			RunStorePath: store.RunStoragePath(),
-		},
-	}
-	sandbox := &Sandbox{
-		ctx: context.Background(),
-		id:  "testSandbox",
-		config: &SandboxConfig{
-			HypervisorConfig: qemuConfig,
-		},
-	}
 
 	// Create the hypervisor fake binary
 	testQemuPath := filepath.Join(testDir, testHypervisor)
 	_, err = os.Create(testQemuPath)
 	assert.NoError(err)
 
 	// Create parent dir path for hypervisor.json
-	parentDir := filepath.Join(store.RunStoragePath(), sandbox.id)
+	parentDir := filepath.Join(store.RunStoragePath(), "testSandbox")
 	assert.NoError(os.MkdirAll(parentDir, DirMode))
 
 	network, err := NewNetwork()
 	assert.NoError(err)
-	err = q.CreateVM(context.Background(), sandbox.id, network, &sandbox.config.HypervisorConfig)
-	assert.NoError(err)
+
+	type testData struct {
+		config      HypervisorConfig
+		expectError bool
+		configMatch bool
+	}
+
+	var data []testData
+
+	// create test configs with default settings
+	for i := 0; i < 18; i++ {
+		data = append(data, testData{newQemuConfig(), false, true})
+	}
+
+	// customize configs
+	data[1].config.SeccompSandbox = "enable=1"
+	data[2].config.InitrdPath = ""
+	data[2].config.ImagePath = testQemuImagePath
+	data[3].config.Debug = true
+	data[4].config.IOMMU = true
+	data[5].config.GuestMemoryDumpPath = "/tmp"
+	data[6].config.DisableGuestSeLinux = false
+	data[6].configMatch = false
+	data[7].config.PCIeRootPort = 1
+	data[8].config.EnableVhostUserStore = true
+	data[8].config.HugePages = true
+	data[9].config.EnableVhostUserStore = true
+	data[9].config.HugePages = false
+	data[9].expectError = true
+	data[10].config.BootToBeTemplate = true
+	data[11].config.BootFromTemplate = true
+	data[12].config.BootToBeTemplate = true
+	data[12].config.SharedFS = config.VirtioFS
+	data[12].expectError = true
+	data[13].config.FileBackedMemRootDir = "/tmp/xyzabc"
+	data[13].config.HugePages = true
+	data[14].config.SharedFS = config.VirtioFS
+	data[15].config.BlockDeviceDriver = ""
+	data[16].config.SharedFS = config.VirtioFSNydus
+	data[17].config.VMid = "testSandbox"
+
+	for i, d := range data {
+		msg := fmt.Sprintf("test[%d]", i)
+
+		q := &qemu{
+			config: HypervisorConfig{
+				VMStorePath:  store.RunVMStoragePath(),
+				RunStorePath: store.RunStoragePath(),
+			},
+		}
+
+		err = q.CreateVM(context.Background(), "testSandbox", network, &d.config)
+
+		if d.expectError {
+			assert.Error(err, msg)
+			continue
+		}
+
+		assert.NoError(err, msg)
+
+		if d.configMatch {
+			assert.Exactly(d.config, q.config, msg)
+		}
+
+		mem := q.GetTotalMemoryMB(context.Background())
+		assert.True(mem > 0)
+
+		err = q.canDumpGuestMemory("/tmp")
```

Comment:
If this dump works, the file in `/tmp` could be very large. Please create a temporary directory for this. The doc below shows how to create a tmpdir that is automatically removed once the test finishes:

- https://github.com/kata-containers/kata-containers/blob/main/docs/Unit-Test-Advice.md#temporary-files

Reply:
// canDumpGuestMemory check if can do a guest memory dump operation.
// for now it only ensure there must be double of VM size for free disk spaces
func (q *qemu) canDumpGuestMemory(dumpSavePath string) error {